lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java

```diff
@@ -200,20 +200,38 @@ public APIGatewayProxyResponseEvent handleRequest(
                     HttpStatusCode.FORBIDDEN,
                     new CommonExpressOAuthError(
                             OAuth2Error.ACCESS_DENIED, customOAuth2ErrorDescription));
+        } catch (NumberFormatException e) {
+            LOGGER.error(String.format("Parsing value of %s to int failed", MAXIMUM_ATTEMPT_COUNT));
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatusCode.INTERNAL_SERVER_ERROR,
+                    new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
         } catch (OAuthHttpResponseExceptionWithErrorBody e) {
             eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
 
-            LOGGER.debug(e.getMessage(), e);
```

Comment:
As we control this exception is not safe to log the whole thing as an error

Reply:
The pattern for exceptions/errors is they are only logged at source and OAuthHttpResponseExceptionWithErrorBody is thrown with a status code and message set, and not logged. If OAuthHttpResponseExceptionWithErrorBody logged here it would double log every error.

If logging is debug level the error is returned in the redirect and will appear in the front end (or lambda reply) -  other wise at info level the spec compliant server_error is used.